matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/DefaultRoomDirectoryService.kt

```diff
@@ -43,4 +46,13 @@ internal class DefaultRoomDirectoryService @Inject constructor(
     override suspend fun setRoomDirectoryVisibility(roomId: String, roomDirectoryVisibility: RoomDirectoryVisibility) {
         setRoomDirectoryVisibilityTask.execute(SetRoomDirectoryVisibilityTask.Params(roomId, roomDirectoryVisibility))
     }
+
+    override suspend fun checkAliasAvailability(aliasLocalPart: String?): Result<Unit> {
+        return try {
+            roomAliasAvailabilityChecker.check(aliasLocalPart)
+            Result.success(Unit)
+        } catch (failure: RoomAliasError) {
+            Result.failure(failure)
+        }
```

Comment:
This is not following the current arch. It's up to the caller to handle the Exception.
If you handle the exception here, maybe this method should return a simple boolean.
Also NetworkError still have to be handled by the caller if it stay like that.

Reply:
yes network erros still have to be handled by the caller.
But here the 'exception' is kind of more the expected result than an exceptional outcome, so wanted to make it obvious instead of letting the caller guessing what to catch

Reply:
As per discussion I created a sealed class AliasAvailabilityResult

Reply:
thx